Neos.ContentRepository.NodeAccess/Classes/FlowQueryOperations/ContextOperation.php

```diff
@@ -0,0 +1,134 @@
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\NodeAccess\FlowQueryOperations;
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\ContentRepository\Core\DimensionSpace\DimensionSpacePoint;
+use Neos\ContentRepository\Core\Feature\SubtreeTagging\Dto\SubtreeTag;
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
+use Neos\ContentRepository\Core\Projection\ContentGraph\VisibilityConstraints;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
+use Neos\Eel\FlowQuery\FlowQuery;
+use Neos\Eel\FlowQuery\FlowQueryException;
+use Neos\Eel\FlowQuery\Operations\AbstractOperation;
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * "context" operation working on ContentRepository nodes. Modifies the ContentRepository Context of each
+ * node in the current FlowQuery context by the given properties and returns the same
+ * nodes by identifier if they can be accessed in the new Context (otherwise they
+ * will be skipped).
+ *
+ * Example:
+ *
+ * 	q(node).context({'invisibleContentShown': true}).children()
+ *
+ * Supported options:
+ * - workspaceName
+ * - dimensions
+ * - invisibleContentShown
+ *
+ * Unsupported legacy options:
+ * - currentDateTime
+ * - targetDimensions
+ * - removedContentShown
+ * - inaccessibleContentShown
+ *
+ */
+class ContextOperation extends AbstractOperation
+{
+    use CreateNodeHashTrait;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @var string
+     */
+    protected static $shortName = 'context';
+
+    /**
+     * {@inheritdoc}
+     *
+     * @var integer
+     */
+    protected static $priority = 1;
+
+    #[Flow\Inject()]
+    protected ContentRepositoryRegistry $contentRepositoryRegistry;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param array<mixed> $context $context onto which this operation should be applied (array or array-like object)
+     * @return boolean true if the operation can be applied onto the $context, false otherwise
+     */
+    public function canEvaluate($context)
+    {
+        return count($context) === 0 || (isset($context[0]) && ($context[0] instanceof Node));
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param FlowQuery $flowQuery The FlowQuery object
+     * @param array<mixed> $arguments The arguments for this operation
+     * @return void
+     * @throws FlowQueryException
+     */
+    public function evaluate(FlowQuery $flowQuery, array $arguments)
+    {
+        if (!isset($arguments[0]) || !is_array($arguments[0])) {
+            throw new FlowQueryException('context() requires an array argument of context properties', 1398030427);
+        }
+
+        $newContextProperties = $arguments[0];
```

Comment:
Just naming, but ... why "new"?

Reply:
these are the things the _new_ subgraph should contain ... basically the delta ... idk naming inspired by the old implementation.